FreeRTOS/FreeRTOS
# Add eTaskConfirmSleepModeStatus unit test for SMP

Description
-----------
This PR is to verify the kernel PR which fix the implementation of `eTaskConfirmSleepModeStatus`.

Single core has `eTaskConfirmSleepModeStatus`. However, the number of idle task is set to a fixed number 1, which is not true for SMP.

#### In this PR
* Add a regression test case with eTaskConfirmSleepModeStatus for SMP to ensure that the implementation is updated for SMP.

Test Steps
-----------
Before the kernel PR fix, the test shows error.
```c
typedef enum
{
    eAbortSleep = 0,           /* A task has been made ready or a context switch pended since portSUPPRESS_TICKS_AND_SLEEP() was called - abort entering a sleep mode. */
    eStandardSleep,            /* Enter a sleep mode that will not last any longer than the expected idle time. */
    #if ( INCLUDE_vTaskSuspend == 1 )
        eNoTasksWaitingTimeout /* No tasks are waiting for a timeout so it is safe to enter a sleep mode that can only be exited by an external interrupt. */
    #endif /* INCLUDE_vTaskSuspend */
} eSleepModeStatus;
```

```
/mnt/c/msys64/home/chinglee/kernel/unit-test/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_mock/covg_multiple_priorities_no_timeslice_mock_utest.c:1307:test_coverage_eTaskConfirmSleepModeStatus_no_tasks_waiting_timeout:FAIL: Expected 2 Was 1

-----------------------
21 Tests 1 Failures 0 Ignored
```

After the kernel PR fix, the test case can be passed.
```
/mnt/c/msys64/home/chinglee/kernel/unit-test/FreeRTOS/FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_mock/covg_multiple_priorities_no_timeslice_mock_utest.c:1283:test_coverage_eTaskConfirmSleepModeStatus_no_tasks_waiting_timeout:PASS

-----------------------
21 Tests 0 Failures 0 Ignored
OK
```

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
